### PR TITLE
Adds upgradeFrom field in templateVersion

### PR DESCRIPTION
**Pull request by mrajashree**
UpgradeFrom field is added to the templateVersion.
While getting the `upgradeVersionLinks` for `currentVersion=v0.2.0`, if `otherVersion=v0.4.0`, `v0.4.0` will be included in `upgradeVersionLinks` only if its `upgradeFrom` field is empty.
If the `upgradeFrom` field for `v0.4.0` is `>=v0.3.0`, then `v0.2.0` will not have `v0.4.0` in its `upgradeVersionLinks` field.

### CODE REVIEW

**Comment by mrajashree** [comment]
@vincent I've added the upgradeFrom field. The build failed due to `Error: missing Docker image`

**Comment by mrajashree** [comment]
@vincent I've added the upgradeFrom field. The build failed due to `Error: missing Docker image`. Can you please take a look?

**Comment by mrajashree** [comment]
@vincent I've added the upgradeFrom field. The build failed due to `Error: missing Docker image`. Can you please take a look?

**Comment by mrajashree** [comment]
@vincent I've added the upgradeFrom field. The build failed due to `Error: missing Docker image`. Can you please take a look?

**Comment by mrajashree** [comment]
@vincent I've added the upgradeFrom field. The build failed due to `Error: missing Docker image`. Can you please take a look?

File: manager/catalog_manager.go [hunk 1]
```diff
@@ -366,14 +367,28 @@ func GetNewTemplateVersions(path string) (model.Template, bool) {
...
-				if err == nil && currentVersion.LessThan(*otherVersion) {
-					copyOfversionLinks[key] = value
+				upgradeRange, err := getUpgradeFrom(&templateOtherMetaData)
+				if err != nil {
+					log.Errorf("Error %v getting semRange ", err)
+					continue
+				}
+
+				if err == nil && currentVersion.LT(*otherVersion) {
+					if upgradeRange != nil {
```
Comment (vincent99) [inline manager/catalog_manager.go]: if upgradeRange == nil || upgradeRange(*currentVersion) {
  copyOfVersionLinks[key] = value
}

File: manager/catalog_manager.go [hunk 2]
```diff
@@ -366,14 +367,28 @@ func GetNewTemplateVersions(path string) (model.Template, bool) {
...
+						if upgradeRange(*currentVersion) {
+							copyOfversionLinks[key] = value
```
Comment (vincent99) [inline manager/catalog_manager.go]: pls change to `copyOfVersionLinks`

Comment (mrajashree) [inline manager/catalog_manager.go]: @vincent99 didn't get that, what should I change?

**Comment by mrajashree** [comment]
@vincent99 made the changes in the condition and changed to copyOfVersionLinks

File: manager/catalog_manager.go [hunk 3]
```diff
@@ -12,10 +12,10 @@ import (
 	"time"
 
 	log "github.com/Sirupsen/logrus"
-	"github.com/coreos/go-semver/semver"
+	"github.com/blang/semver"
 	"github.com/rancher/go-rancher/client"
 	"github.com/rancher/rancher-catalog-service/model"
-	"gopkg.in/yaml.v2"
+	// "gopkg.in/yaml.v2"
```
Comment (ibuildthecloud) [inline manager/catalog_manager.go]: @mrajashree Can you remove this comment?

**Comment by mrajashree** [comment]
@ibuildthecloud removed the comment

**Pull request merged**